api/app/Models/PoolCandidate.php

```diff
@@ -844,32 +881,47 @@ public function scopeOrderByPoolName(Builder $query, ?array $args): Builder
         return $query;
     }
 
-    public function scopeOrderByClaimVerification(Builder $query, ?string $sortOrder)
+    public function scopeOrderByClaimVerification(Builder $query, ?array $args)
     {
-        $orderWithoutDirection = '
-                    CASE
-                    WHEN priority_verification=\'ACCEPTED\' OR priority_verification=\'UNVERIFIED\' then 40
-                    WHEN (veteran_verification=\'ACCEPTED\' OR veteran_verification=\'UNVERIFIED\') AND (priority_verification IS NULL OR priority_verification=\'REJECTED\') then 30
-                    WHEN (users.citizenship=\'CITIZEN\' OR users.citizenship=\'PERMANENT_RESIDENT\') AND (priority_verification IS NULL OR priority_verification=\'REJECTED\') AND (veteran_verification IS NULL OR veteran_verification=\'REJECTED\') then 20
-                    else 10
-                    END';
 
-        if ($sortOrder && $sortOrder == 'DESC') {
-            $order = $orderWithoutDirection.' DESC';
+        if (isset($args['order'])) {
+
+            $orderWithoutDirection = "
+        CASE
+            WHEN
+                (priority_verification = 'ACCEPTED' OR priority_verification = 'UNVERIFIED')
+            THEN
+                40
+            WHEN
+                (veteran_verification = 'ACCEPTED' OR veteran_verification = 'UNVERIFIED')
+                AND
+                (priority_verification IS NULL OR priority_verification = 'REJECTED')
+            THEN
+                30
+            WHEN
+                (users.citizenship = 'CITIZEN' OR users.citizenship = 'PERMANENT_RESIDENT')
+                AND
+                (priority_verification IS NULL OR priority_verification = 'REJECTED')
+                AND
+                (veteran_verification IS NULL OR veteran_verification = 'REJECTED')
+            THEN
+                20
+            ELSE
+                10
+        END
+        ";
 
             $query
                 ->join('users', 'users.id', '=', 'pool_candidates.user_id')
-                ->select('users.citizenship', 'pool_candidates.*')
-                ->orderBy('is_bookmarked', 'DESC')
-                ->orderByRaw($order);
-        } elseif ($sortOrder && $sortOrder == 'ASC') {
-            $order = $orderWithoutDirection.' ASC';
+                ->select('users.citizenship', 'pool_candidates.*');
 
-            $query
-                ->join('users', 'users.id', '=', 'pool_candidates.user_id')
-                ->select('users.citizenship', 'pool_candidates.*')
-                ->orderBy('is_bookmarked', 'DESC')
-                ->orderByRaw($order);
+            if (isset($args['useBookmark']) && $args['useBookmark']) {
+                $query->orderBy('is_bookmarked', 'DESC');
+            }
+
+            $order = sprintf('%s %s', $orderWithoutDirection, $args['order']);
+
+            $query->orderByRaw($order);
```

Comment:
This sorts randomly for candidates who match both. Should we add a tertiary sort to keep it stable? I was thinking `updated_at` :thinking:

Reply:
Right.. we should break the tie with additional critiera .. I would like to think as who applied first should go first when there is a tie.. like submitted_at ascending. May be @desfam ?